io/azure_integration_test.go

```diff
@@ -0,0 +1,148 @@
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+//go:build integration
+
+package io_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob"
+	"github.com/Azure/azure-sdk-for-go/sdk/storage/azblob/bloberror"
+	"github.com/apache/iceberg-go"
+	"github.com/apache/iceberg-go/catalog"
+	sqlcat "github.com/apache/iceberg-go/catalog/sql"
+	"github.com/apache/iceberg-go/io"
+	"github.com/stretchr/testify/suite"
+	"github.com/uptrace/bun/driver/sqliteshim"
+	"gocloud.dev/blob/azureblob"
+)
+
+const (
+	accountName      = "devstoreaccount1"
+	accountKey       = "Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw=="
+	endpoint         = "127.0.0.1:10010"
+	protocol         = "http"
+	containerName    = "warehouse"
+	connectionString = "DefaultEndpointsProtocol=http;AccountName=devstoreaccount1;AccountKey=Eby8vdM02xNOcqFlqUwJPLlmEtlCDXJ1OUzFT50uSRZ6IFsuFq2UVErCz4I6tq/K1SZFPTOtr/KBHBeksoGMGw==;BlobEndpoint=http://127.0.0.1:10010/devstoreaccount1;QueueEndpoint=http://127.0.0.1:10001/devstoreaccount1;TableEndpoint=http://127.0.0.1:10002/devstoreaccount1;"
+)
+
+type AzureBlobIOTestSuite struct {
+	suite.Suite
+
+	ctx context.Context
+}
+
+func (s *AzureBlobIOTestSuite) SetupTest() {
+	s.ctx = context.Background()
+
+	err := createContainerIfNotExist(containerName)
+	s.NoError(err)
+}
+
+func (s *AzureBlobIOTestSuite) TestAzureBlobWarehouseKey() {
+	path := "iceberg-test-azure/test-table-azure"
+	containerName := "warehouse"
+
+	cat, err := catalog.Load(context.Background(), "default", iceberg.Properties{
+		"uri":                       ":memory:",
+		sqlcat.DriverKey:            sqliteshim.ShimName,
+		sqlcat.DialectKey:           string(sqlcat.SQLite),
+		"type":                      "sql",
+		io.AdlsSharedKeyAccountName: accountName,
+		io.AdlsSharedKeyAccountKey:  accountKey,
+		io.AdlsEndpoint:             endpoint,
+		io.AdlsProtocol:             protocol,
+	})
+	s.NoError(err)
+	s.NotNil(cat)
+
+	c := cat.(*sqlcat.Catalog)
+	s.NoError(c.CreateNamespace(s.ctx, catalog.ToIdentifier("iceberg-test-azure"), nil))
+
+	tbl, err := c.CreateTable(s.ctx,
+		catalog.ToIdentifier("iceberg-test-azure", "test-table-azure"),
+		iceberg.NewSchema(0, iceberg.NestedField{
+			Name: "id", Type: iceberg.PrimitiveTypes.Int32, Required: true, ID: 1,
+		}), catalog.WithLocation(fmt.Sprintf("abfs://%s/iceberg/%s", containerName, path)))
+	s.NoError(err)
+	s.NotNil(tbl)
```

Comment:
should add a test here afterwards to load the table and ensure the data matches. Right now you're only testing that there's no errors when it tries to perform the write, there's no confirmation that it actually wrote the right things by attempting to read it back.

Reply:
done.